lib/util/promises.ts

```diff
@@ -32,7 +24,16 @@ function handleError(err: any): never {
     throw error;
   }
 
-  throw err;
+  throw new AggregateError(errors);
+}
+
+function handleError(err: any): never {
+  if (!(err instanceof AggregateError)) {
```

Comment:
We need to be careful, `AggregateError`  is a javascript build-in too since node v15[^1].

[^1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AggregateError

Reply:
In this context, we care only about the `AggregateError` variant thrown by `p-map`/`p-all`